nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/test/java/org/apache/nifi/excel/TestExcelReader.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.util.MockConfigurationContext;
+import org.apache.nifi.util.MockControllerServiceInitializationContext;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.Date;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@ExtendWith(MockitoExtension.class)
+public class TestExcelReader {
+
+    private static final ByteArrayOutputStream ALL_TYPES_CONTENTS = new ByteArrayOutputStream();
+    private static final Map<String, Class<?>> FIELD_NAMES_AND_CLASS_TYPES = new LinkedHashMap<>();
+
+    @Mock
+    private ComponentLog logger;
+
+    private ExcelReader excelReader;
+    private Map<PropertyDescriptor, String> properties;
+
+    @BeforeAll
+    static void setUpBeforeAll() throws Exception {
+        InputStream inputStream = Files.newInputStream(Paths.get("src/test/resources/excel/all_types.xlsx"));
+        IOUtils.copyLarge(inputStream, ALL_TYPES_CONTENTS);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Date_Custom", Date.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Time_Custom", Time.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Timestamp", Timestamp.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Boolean_Function", Boolean.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Number_Function", Integer.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Blank_Cell", String.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("String", String.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Numeric", Float.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Variable_Missing", String.class);
+        FIELD_NAMES_AND_CLASS_TYPES.put("Syntax_Error", String.class);
+    }
+
+    @BeforeEach
+    void setUp() {
+        excelReader = new ExcelReader();
+        properties = new HashMap<>();
+        new ExcelReader().getSupportedPropertyDescriptors().forEach(prop -> properties.put(prop, prop.getDefaultValue()));
+    }
+
+    @Test
+    void testReadWithInferredSchema() throws Exception {
+        properties.put(DateTimeUtils.DATE_FORMAT, "M/dd/yy");
+        properties.put(DateTimeUtils.TIME_FORMAT, "HH:mm");
+        properties.put(DateTimeUtils.TIMESTAMP_FORMAT, "M/dd/yyyy HH:mm:ss");
```

Comment:
It would be helpful to declare static values for the date, time, and timestamp formats, which can be reused across this class.